static/absolute/js/searchPage.js

```diff
@@ -24,14 +26,35 @@ window.onload = function () {
         return;
       }
     } else {
-      // TODO(ved): Use API to convert address --> long/lat
-      alert("This is not yet supported");
-      return;
+      if (/[`!#$%\^&*+=\-\[\]\\';/{}|\\":<>\?]/.test(address)) {
+        alert(
+          "Invalid characters were entered with the address. Please remove them and try again."
+        );
+        return;
+      }
+
+      let formattedAddress = address.replace(" ", "%20");
+      formattedAddress = formattedAddress.replace(",", "%2C");
+      const coords = await (
+        await fetch(`/api/${formattedAddress}/geocode`)
+      ).json();
+
+      if (coords.status === 200) {
+        lat = coords.data.lat;
+        long = coords.data.lng;
+      } else {
+        alert(
+          "We could not find the latitude and longitude of that address." +
+            "Please try again, and make sure to follow the address guidelines."
+        );
+        return;
+      }
     }
 
     let data;
```

Comment:
It would be nice to specify which data this is. The name data is a little vague

Reply:
I know this PR is a work in progress, but  I can't find anywhere the variable data is used in this file.

Reply:
Oh, I see this was not your change.

Reply:
This is not one of my changes, but I can change the variable name. 

'data' is being used in the section right below it. It holds the json response from the post method used to add user information to the database. It holds the status to show whether or not information was added to datastore successfully.

Reply:
I changed this variable name to 'postResponse', which is a bit more descriptive.